### DISCUSSION TITLE
Paul Bigger - Darklang

### DISCUSSION THREAD

**Discussion by just-be-dev** [devtools-fm/devtools.fm#108] [Announcements]
## Background

[Twitter](https://twitter.com/paulbiggar)
[GitHub](https://github.com/pbiggar)
[Darklang](https://darklang.com/)
[Darklang GitHub](https://github.com/darklang/dark)

## Themes

- DevTools / Productivity
- Visual Programming
- Functional Programming
- Startups

## Intro

Andrew: Hello, welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin.

Justin: Hey everyone! Our guest today is Paul Bigger. Paul was the founder of CircleCI and spends his time these days working on Dark, an integrated development environment for building serverless backends. I'm probably not really doing Dark justice with that description, ha. Before we dig more into that, Paul, would you like to tell our audience a little more about yourself?

### CircleCI

- [x] What prompted you to found CircleCI? Was it your first startup?
- [x] CircleCi has been successful. What was the experience like from shifting from a founder/person running the company to a board member?
- [x] Do you still use CircleCI?

### Dark Lang

- [x] How would you describe dark to someone who has never heard of it?
- [x] What inspired you to build dark?
- [x] How does building a service with dark compare to more traditional methods?
- [x] Does it work with git? How are merge conflicts resolved?
- [x] Creating a dark application is very visual. What are some cool features that this visual-ness unlocks? What challenges does it surface?
- [x] Dark is it's own language? What have some challenges been writing a new language? 
- [x] How much though goes into the DX of the language itself? What other languages might it compare to?
- [x] does dark provide and kind of type safety?
- [x] How does testing work?
- [x] What tools do you provide for debugging applications?
- [x] Does dark take care of infra and scaling for me? Is basically everything abstracted away?
- [x] Dark comes with a package manager? what kind of code can I pull in with it?
- [x] Is it only serverless functions? Can I create long running servers?
- [x] What does dark compile down too? 
- [x] You recently went through a large rewrite of dark. What did you rewrite and why did you do it? Was it successful? 
- [x] What are some of your favorite features built into dark?
- [x] Do you run your own cloud or is it built on something like AWS?

### Future

- [x] Who should be using dark? What type of apps find the most success with it?
- [x] How do you plan to monetize dark?
- [x] (If it's hosting a dark server) Can I host it myself?
- [x] What big features are your working towards to get dark to a v2?
- [x] What do you hope the future of programming looks like?

**Comment by hipstersmoothie** [comment]
My tooltips:

- https://liblab.com
- https://github.com/module-federation/nextjs-mf

**Comment by just-be-dev** [comment]
Tooltips:
- https://github.com/harc/ohm
- https://github.com/valence-rs/valence

**Comment by pbiggar** [comment]
Tooltips: 
- https://tailwindcss.com/
- https://www.lg.com/us/monitors/lg-49WL95C-W-ultrawide-monitor
- https://docs.microsoft.com/en-us/dotnet/fsharp/what-is-fsharp
- https://www.zsa.io/moonlander/
